### ISSUE TITLE
Users should be able to run cuppa while offline efficiently

### ISSUE THREAD

**Issue by ja11sop** [ja11sop/cuppa#9]
Right now being offline results in slow and apparently broken behaviour. Being offline should be detected and made explicit so that online requirements can be circumvented or at least accurately reported on. It should be possible to also explicitly select an `--offline` mode if it is know that internet access is not needed or should not be attempted.

**Comment by feedhandler** [comment]
Piggybacking on this issue: we're behind a proxy and cuppa is hanging in check_current_version():
```
dmsdev@dmsdev> scons
scons: Reading SConscript files ...
cuppa: version: [info] cuppa: version 0.9.23
......hangs here
```
The next source code line after that is         
```python
pypi = xmlrpclib.ServerProxy('http://pypi.python.org/pypi')
```
so I'm guessing it is stuck somewhere in there

**Comment by ja11sop** [comment]
This should work on `master` at the moment but it's not yet released - you could try installing from the repo or better yet clone locally and `pip install -e` from that local repo.

**Comment by ja11sop** [comment]
This should be fixed in `0.9.24`
